dsp-user/core/src/test/scala/dsp/user/domain/UserDomainSpec.scala

```diff
@@ -4,3 +4,126 @@
+        assertTrue(updatedUser.status == user.status)
+    },
+    test("update the givenName") {
+      val user     = SharedTestData.simpleUser1
+      val newValue = GivenName.make("newGivenName").fold(e => throw e.head, v => v)
+      for {
+        updatedUser <- ZIO.succeed(user.updateGivenName(newValue))
+      } yield assertTrue(updatedUser.givenName == newValue) &&
+        assertTrue(updatedUser.email == user.email) &&
+        assertTrue(updatedUser.givenName != user.givenName) &&
+        assertTrue(updatedUser.familyName == user.familyName) &&
+        assertTrue(updatedUser.password == user.password) &&
+        assertTrue(updatedUser.language == user.language) &&
+        assertTrue(updatedUser.status == user.status)
+    },
+    test("update the familyName") {
+      val user     = SharedTestData.simpleUser1
+      val newValue = FamilyName.make("newFamilyName").fold(e => throw e.head, v => v)
+      for {
+        updatedUser <- ZIO.succeed(user.updateFamilyName(newValue))
+      } yield assertTrue(updatedUser.familyName == newValue) &&
+        assertTrue(updatedUser.email == user.email) &&
+        assertTrue(updatedUser.givenName == user.givenName) &&
+        assertTrue(updatedUser.familyName != user.familyName) &&
+        assertTrue(updatedUser.password == user.password) &&
+        assertTrue(updatedUser.language == user.language) &&
+        assertTrue(updatedUser.status == user.status)
+    },
+    test("update the password") {
+      val user     = SharedTestData.simpleUser1
+      val newValue = PasswordHash.make("newPassword1").fold(e => throw e.head, v => v)
+      for {
+        updatedUser <- ZIO.succeed(user.updatePassword(newValue))
+      } yield assertTrue(updatedUser.password == newValue) &&
+        assertTrue(updatedUser.email == user.email) &&
+        assertTrue(updatedUser.givenName == user.givenName) &&
+        assertTrue(updatedUser.familyName == user.familyName) &&
+        assertTrue(updatedUser.password != user.password) &&
+        assertTrue(updatedUser.language == user.language) &&
+        assertTrue(updatedUser.status == user.status)
+    },
+    test("update the language") {
+      val user     = SharedTestData.simpleUser1
+      val newValue = LanguageCode.make("fr").fold(e => throw e.head, v => v)
+      for {
+        updatedUser <- ZIO.succeed(user.updateLanguage(newValue))
+      } yield assertTrue(updatedUser.language == newValue) &&
+        assertTrue(updatedUser.email == user.email) &&
+        assertTrue(updatedUser.givenName == user.givenName) &&
+        assertTrue(updatedUser.familyName == user.familyName) &&
+        assertTrue(updatedUser.password == user.password) &&
+        assertTrue(updatedUser.language != user.language) &&
+        assertTrue(updatedUser.status == user.status)
+    },
+    test("update the status") {
+      val user     = SharedTestData.simpleUser1
+      val newValue = UserStatus.make(false).fold(e => throw e.head, v => v)
```

Comment:
That might make sense, but names `ThrowOrGet`? Or maybe there is better way of doing it than `fold`.

Reply:
You shouldn't throw in the first place (but I think that was discussed yesterday). What error will the domain entity return if the update is not possible? It would probably also make sense to return a Validation. In this case, you could have both steps inside a for comprehension, which should clean it up nicely.

Here is an example of what I mean: https://zio.github.io/zio-prelude/docs/functionaldatatypes/validation#chaining-validations

After that you can transform the Validation in whatever shape is needed: https://zio.github.io/zio-prelude/docs/functionaldatatypes/validation#extracting-validation-values

It should never be necessary to use `fold`. Using `fold` is a sign, that something is not quite right.

Reply:
I make the `User.make(...)` also return a `Validation` as you suggested and removed all the `fold(...)`s from the tests. Could you check if it is as you imagined?